kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/system/SystemServiceImpl.java

```diff
@@ -1138,8 +1136,7 @@ private static void createDirIfNotExists(String fileName)
 		// Make sure the configuration directory exists - create it if not		
 		File file = new File(fileName);
 		if(!file.exists() && !file.mkdirs()) {
-			s_logger.error("Failed to create the temporary configuration directory: " + fileName);
-			System.exit(-1);
+			s_logger.error("Failed to create the temporary configuration directory: {}", fileName);
```

Comment:
@ctron Can you just throw an exception here instead? We should not continue the flow of the execution here.